frontend/src/screens/wallet/index.tsx

```diff
@@ -8,28 +8,73 @@ import { Link } from "react-router-dom";
 import AppHeader from "src/components/AppHeader";
 import BreezRedeem from "src/components/BreezRedeem";
 import ExternalLink from "src/components/ExternalLink";
+import { SelfCustodyIcon } from "src/components/icons/SelfCustodyIcon";
 import Loading from "src/components/Loading";
 import TransactionsList from "src/components/TransactionsList";
+import { TransferFundsButton } from "src/components/TransferFundsButton";
 import {
   Alert,
   AlertDescription,
   AlertTitle,
 } from "src/components/ui/alert.tsx";
 import { Button } from "src/components/ui/button";
+import { ALBY_HIDE_HOSTED_BALANCE_BELOW as ALBY_HIDE_HOSTED_BALANCE_LIMIT } from "src/constants.ts";
+import { useAlbyBalance } from "src/hooks/useAlbyBalance";
 import { useBalances } from "src/hooks/useBalances";
+import { useChannels } from "src/hooks/useChannels";
 import { useInfo } from "src/hooks/useInfo";
 
 function Wallet() {
   const { data: info, hasChannelManagement } = useInfo();
   const { data: balances } = useBalances();
+  const { data: channels } = useChannels();
+  const { data: albyBalance, mutate: reloadAlbyBalance } = useAlbyBalance();
 
   if (!info || !balances) {
     return <Loading />;
   }
 
+  const showMigrateCard =
+    albyBalance && albyBalance.sats > ALBY_HIDE_HOSTED_BALANCE_LIMIT;
+
   return (
     <>
       <AppHeader title="Wallet" description="" />
+      {showMigrateCard && (
+        <div className="flex flex-1 items-center justify-center rounded-lg border border-dashed shadow-sm p-8">
+          <div className="flex flex-col items-center gap-1 text-center max-w-md">
+            <SelfCustodyIcon
+              width="211"
+              height="64"
+              viewBox="0 0 211 64"
+              strokeWidth="0"
+              className="text-primary-background"
+            />
+            <h3 className="mt-4 text-lg font-semibold">
+              Your funds ({new Intl.NumberFormat().format(albyBalance.sats)}{" "}
+              sats) are still hosted by Alby.
+            </h3>
+            <p className="text-sm text-muted-foreground mb-4">
+              {channels && channels.length > 0
+                ? "Transfer funds from your Alby hosted balance to your self-custodial wallet."
+                : "Migrate funds from your Alby hosted balance to start using your self-custodial wallet."}
+            </p>
+            {channels && channels.length > 0 ? (
+              <TransferFundsButton
+                channels={channels}
```

Comment:
you could have used the SWR functions within the component rather than passing these as args. But I think it's ok